spec/support/shared_examples_for_otp_delivery_preference_validation.rb

```diff
@@ -0,0 +1,24 @@
+shared_examples 'an otp delivery preference form' do
+  let(:phone) { '555-555-5000' }
+
+  before do
+    allow_any_instance_of(UpdateUser).to receive(:call)
```

Comment:
We discourage the use of `allow_any_instance_of`, as do the RSpec docs: https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance

The problem with this before statement is that it makes it look like `UpdateUser` should be called even if the validation failed, which is not true.

Instead, we do something like:
```ruby
update_user = instance_double(UpdateUser)
attributes = { otp_delivery_preference: 'voice' }
allow(UpdateUser).to receive(:new).with(user: user, attributes: attributes).and_return(update_user)
expect(update_user).to receive(:call) # when validation passes

capabilities = instance_double(PhoneNumberCapabilities)
allow(PhoneNumberCapabilities).to receive(:new).with(phone).and_return(capabilities)
allow(capabilities).to receive(:sms_only).and_return(false)

result = subject.submit(phone: phone, otp_delivery_preference: 'voice')
```
and
```ruby
expect(update_user).to_not receive(:call) # when validation fails
```